### PR TITLE
Project ARES: Print ASRS Log

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request

Allows the ASRS Audit Log to be printed from the ARES Interface.

# Explain why it's good for the game

Many times I've seen CMB request access to the audit log for the purposes of checking on black market activity. Whilst this is quite possible, there's no simple way to take a copy of those records. Now there is.

# Testing Photographs and Procedure
(This image has a bug in it but that has been fixed)
image

# Changelog
:cl:
add: Added a print button to the ASRS audit log.
/:cl:

### CODE REVIEW

**Comment by vero5123** [comment]
This is mad copy pasta, I know because I copy pastad the same thing for one of my prs.

**Review by vero5123 — COMMENTED** [review]
nitpicks aside, maybe it would be best if there was a cooldown on the print option.

https://github.com/cmss13-devs/cmss13/assets/73014819/5ba15c68-8c10-4068-906d-671b761fbe28

File: code/game/machinery/ARES/ARES_interface.dm [hunk 1]
```diff
@@ -337,6 +341,74 @@
...
+							<th scope="row">[req_order.time]</th>
+							<td>[req_order.user]</td>
+							<td>[req_order.title]</td>
+							<td>[req_order.details]</td>
+							</tr>
+							"}
+
+			contents += "</center></tbody></table>"
+
+			var/obj/item/paper/log = new /obj/item/paper(src.loc)
```
Comment (vero5123) [inline code/game/machinery/ARES/ARES_interface.dm:406]:
```suggestion
			var/obj/item/paper/log = new(loc)
```

File: code/game/machinery/ARES/ARES_interface.dm [hunk 2]
```diff
@@ -337,6 +341,74 @@
 			last_menu = current_menu
 			current_menu = "core_security"
 
+		// -- Print ASRS Audit Log -- //
+		if("print_req")
+			playsound = FALSE
+			if(!COOLDOWN_FINISHED(src, printer_cooldown))
+				playsound(src, 'sound/machines/buzz-two.ogg', 15, 1)
+				return FALSE
+			if(!datacore.records_asrs.len)
```
Comment (vero5123) [inline code/game/machinery/ARES/ARES_interface.dm:350]:
```suggestion
			if(!length(datacore.records_asrs))
```

File: code/game/machinery/ARES/ARES_interface.dm [hunk 3]
```diff
@@ -337,6 +341,74 @@
...
+							<th scope="col">Source</th>
+							<th scope="col">Order</th>
+							</tr>
+						</thead>
+						<tbody>
+						"}
+
+			for(var/datum/ares_record/requisition_log/req_order as anything in datacore.records_asrs)
+				if(!istype(req_order))
+					continue
```
Comment (vero5123) [inline code/game/machinery/ARES/ARES_interface.dm:393]:
```suggestion
			for(var/datum/ares_record/requisition_log/req_order as anything in datacore.records_asrs)
```
as anything removes the implicit type check, if the only thing inside of records_asrs is of type datum/ares_record/requisition_log/ then you don't need to do type check.

**Review by vero5123 — COMMENTED** [review]
nitpicks

File: code/game/machinery/ARES/ARES_interface.dm [hunk 4]
```diff
@@ -337,6 +341,72 @@
...
+							</tr>
+							"}
+
+			contents += "</center></tbody></table>"
+
+			var/obj/item/paper/log = new(loc)
+			log.name = "ASRS Audit Log"
+			log.info += contents
+			log.icon_state = "paper_uscm_words"
+			visible_message(SPAN_NOTICE("\The [src] prints out a paper."))
```
Comment (harryob) [inline code/game/machinery/ARES/ARES_interface.dm:408]:
```suggestion
			visible_message(SPAN_NOTICE("[src] prints out a paper."))
```

**Pull request merged**